server/service/datapool/download_data_handler.go

```diff
@@ -1,10 +1,70 @@
 package datapool
 
 import (
+	"fmt"
 	"net/http"
+	"strconv"
+
+	"github.com/gorilla/mux"
+
+	datapooltypes "github.com/medibloc/panacea-core/v2/x/datapool/types"
+	"github.com/medibloc/panacea-data-market-validator/types"
+
+	log "github.com/sirupsen/logrus"
 )
 
-func (svc *dataPoolService) handleDownloadData(writer http.ResponseWriter, request *http.Request) {
-	//TODO implement me
-	panic("implement me")
+func (svc *dataPoolService) handleDownloadData(w http.ResponseWriter, r *http.Request) {
+	if err, errStatusCode := validateDownloadRequest(r); err != nil {
+		log.Errorf("invalid download request: %v", err)
+		http.Error(w, err.Error(), errStatusCode)
+		return
+	}
+
+	redeemer := r.FormValue("requester_address")
+
+	poolID, err := strconv.ParseUint(mux.Vars(r)[types.PoolIDKey], 10, 64)
+	if err != nil {
+		log.Errorf("invalid pool ID: %v", err)
+		http.Error(w, "invalid pool ID", http.StatusBadRequest)
+		return
+	}
+
+	redeemHistory, err := svc.PanaceaClient.GetDataPassRedeemHistory(redeemer, poolID)
+	if err != nil {
+		log.Errorf("failed to get redeem receipt: %v", err)
+		http.Error(w, "failed to get data pass redeem receipt", http.StatusInternalServerError)
+		return
+	}
+
+	if len(redeemHistory.DataPassRedeemReceipts) == 0 {
+		log.Errorf("redeem receipt not found under %s", redeemHistory.Redeemer)
+		http.Error(w, "redeem receipt not found", http.StatusNotFound)
+		return
+	}
+
+	redeemedRound := findMaxRound(redeemHistory.DataPassRedeemReceipts)
+
+	fmt.Print(poolID, redeemedRound)
```

Comment:
This print line will be removed in the next PR :)

Reply:
When the print line is removed, will the function `findMaxRound` will be removed too?

Reply:
No. 
This query returns a list of redeem receipts redeemed by a redeemer. We need to find the latest round redeemed, and return all the corresponding data.
The `findMaxRound`, which is to get the latest redeemed round, seems little bit confusing so I'll rename it.
maybe.. `getRedeemedRound`?

Reply:
Oh, I understand. I commented because the `getRedeemedRound` is only used in `handleDownloadData`. So it will be used in near future. Did I understand right?